client/state/sites/selectors.js

```diff
@@ -550,3 +554,304 @@ export function getSitePostsPage( state, siteId ) {
 export function getSiteFrontPageType( state, siteId ) {
 	return getSiteOption( state, siteId, 'show_on_front' );
 }
+
+/**
+ * Determines if a Jetpack site has opted in for full-site management
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {?Boolean} if the site can be managed from calypso
+ */
+export function canJetpackSiteManage( state, siteId ) {
+	// for versions 3.4 and higher, canManage can be determined by the state of the Manage Module
+	const siteJetpackVersion = getSiteOption( state, siteId, 'jetpack_version' );
+
+	if ( ! siteJetpackVersion ) {
+		return null;
+	}
+
+	if ( versionCompare( siteJetpackVersion, '3.4', '>=' ) ) {
+		// if we haven't fetched the modules yet, we default to true
+		const isModuleActive = isJetpackModuleActive( state, siteId, 'manage' );
+		return isModuleActive === null ? true : isModuleActive;
+	}
+	// for version lower than 3.4, we cannot not determine canManage, we'll assume they can
+	return true;
+}
+
+/**
+ * Determines if a Jetpack site can update its files.
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site can update its file
+ */
+export function canJetpackSiteUpdateFiles( state, siteId ) {
+	if ( ! isJetpackSite( state, siteId ) ) {
+		return null;
+	}
+
+	if ( ! siteHasMinimumJetpackVersion( state, siteId ) ) {
+		return false;
+	}
+
+	const isMultiNetwork = getSiteOption( state, siteId, 'is_multi_network' );
+
+	if ( isMultiNetwork ) {
+		return false;
+	}
+
+	if ( ! isJetpackSiteMainNetworkSite( state, siteId ) ) {
+		return false;
+	}
+
+	const fileModDisabled = getSiteOption( state, siteId, 'file_mod_disabled' );
+
+	if ( ! fileModDisabled ) {
+		return true;
+	}
+
+	return (
+		! includes( fileModDisabled, 'disallow_file_mods' ) &&
+		! includes( fileModDisabled, 'has_no_file_system_write_access' )
+	);
+}
+
+/**
+ * Determines if a Jetpack site can auto update its files.
+ * This function checks if the given Jetpack site can update its files and if the automatic updater is enabled
+ *
+ * @param {Object} state Global state tree
+ * @param {Number} siteId Site ID
+ * @return {Boolean} true if the site can auto update
+ */
+export function canJetpackSiteAutoUpdateFiles( state, siteId ) {
```

Comment:
If we decide to go with `null` returns, this function should probably return `null`.